armor-write/src/main/java/com/rapid7/armor/store/S3WriteStore.java

```diff
@@ -61,6 +64,12 @@
   private final ShardStrategy shardStrategy;
   private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
   private static final String INTERVAL_TAG = "interval";
+  private static final String TENANT_CACHE_KEY = "tenants";
+  private Cache<String, Set<String>> tenantCache = CacheBuilder.newBuilder()
```

Comment:
Lets avoid using guava. Lets use a a map of SoftReferences.
It doesn't have to be as complicated as this the link below but it should give you some idea on it.

https://blog.shiftleft.io/understanding-jvm-soft-references-for-great-good-and-building-a-cache-244a4f7bb85d

We don't need to worry about it consuming too much memory. Right now we have like under 20K orgs, so 20K strings isn't a lot. If you really want to just make it simply have a persistent in memory cache using a set.